### PR TITLE
Assets Package

### PR DESCRIPTION
**Pull request by GoodBoyDigital**
##### Description of change
Shiny new Assets Manager

This is draft as still got a bunch of stuff to tie up
but made it visible for peeps!
 
##### Pre-Merge Checklist
- [x] Tests and/or benchmarks are included
- [x] Documentation is changed or added
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

### CODE REVIEW

File: packages/assets/src/loader/Loader.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+    /**
+     * Use this to add any plugins to the loadAssets function to use
+     *
+     * @param newParsers - a array of parsers to add to the loader, or just a single one
+     */
+    public addParser(...newParsers: LoaderParser[]): void
+    {
+        this.plugins.push(...newParsers);
+    }
```
Comment (andrewstart) [inline packages/assets/src/loader/Loader.ts:47]: Suggestion: standardize on either "plugins" or "parsers", instead of mixing terms.

**Comment by bigtimebuddy** [comment]
My biggest concern is that I'd like to introduce this in v6 and then make it the default in v7 (removing loaders). Can we make this package optional in v6 and move this PR to `dev`?

File: packages/assets/src/loader/Loader.ts (hunk 1 above)
Comment (GoodBoyDigital) [reply packages/assets/src/loader/Loader.ts:47]: thanks @andrewstart ! yeah, that some legacy bits there! going with parsers!

**Comment by miltoncandelero** [comment]
Is this the end of the TextureCache? 🤔

**Comment by GoodBoyDigital** [comment]
closing as pushing stuff again onto dev..

**Comment by GoodBoyDigital** [comment]
replaced with: https://github.com/pixijs/pixijs/pull/8365

**Pull request closed without merging**